File: classes/street.py
class Street:
    def __init__(self,start, end, name, length, id):
        self.start = start
        self.end = end
        self.name = name
        self.id = id
        self.length = length

        ###end for debugging purposes
        self.passed = 0
        self.entered = 0



    def carEnter(self):
        self.entered += 1
        #self.queue.append(car_id)
        return self.entered - 1

    def carLeave(self):
        self.passed += 1
        return self.passed

    def is_empty(self):
        return self.passed == self.entered


    def reset(self):
        self.passed = 0
        self.entered = 0

File: CPP/street.cpp
#include "street.h"

Street:: Street( int id, int start, int end, int length){
    this->id = id;
    this->start = start;
    this->end = end;
    this->length = length;
}

int Street::carEnter(){
        this->entered += 1;
        return this->entered - 1;
}

int Street::carLeave(){
        this->passed += 1;
        return this->passed;
}

bool Street::is_empty(){
    return this->entered == this->passed;
}

void Street::reset(){
    this->entered = 0;
    this->passed = 0;
}
